.github/workflows/build.yml

```diff
@@ -121,9 +121,42 @@ jobs:
             fetch-upstream: "true"
       fail-fast: false
     steps:
+      - run: |
+          echo "OS_ARCH=`uname -m`" >> $GITHUB_ENV
       - uses: actions/checkout@v3
+      # Cache `cargo-make`, `cargo-cache`, `cargo-sweep`
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}
+      # Cache first-party rust code crate dependencies
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}-${{ hashFiles('.github/workflows/build.yml') }}-${{ github.head_ref }}
+          restore-keys: |
+            ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}-${{ hashFiles('.github/workflows/build.yml') }}
```

Comment:
Similar to the above, to simplify, why not just make the access key:
```
{{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}
```
so it can be used across many different PRs or many different runs?

Reply:
See https://github.com/bottlerocket-os/bottlerocket/pull/2649#discussion_r1044863100

Reply:
Resolving this convo and focusing on above thread ^